[rhoai-2.24] backport changes from pr:1318 to main

Shafi-Hussain opened the pull request:
Updated Dockerfile.konflux.cpu for jupyter-minimal images

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Improvements**
  * Enhanced PDF export for Jupyter notebooks with optimized dependency installation and support across multiple processor architectures.

<!-- review_stack_entry_start -->

[![Review Change Stack](https://storage.googleapis.com/coderabbit_public_assets/review-stack-in-coderabbit-ui.svg)](https://app.coderabbit.ai/change-stack/red-hat-data-services/notebooks/pull/1490?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

<!-- review_stack_entry_end -->

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Shafi-Hussain committed:
backport changes from pr:1318 to main

Signed-off-by: Md. Shafi Hussain <Md.Shafi.Hussain@ibm.com>

Shafi-Hussain committed:
enable verbose logging for build and update ghc settings

Signed-off-by: Md. Shafi Hussain <Md.Shafi.Hussain@ibm.com>

Shafi-Hussain committed:
verbose logging with updated cabal

Signed-off-by: Md. Shafi Hussain <Md.Shafi.Hussain@ibm.com>

Shafi-Hussain commented:
/build-konflux

Shafi-Hussain committed:
verbose logging with updated cabal

Signed-off-by: Md. Shafi Hussain <Md.Shafi.Hussain@ibm.com>

Shafi-Hussain commented:
/build-konflux

puneetsharma21 committed:
install from EPEL and symlink into /usr/local/pandoc/bin

Shafi-Hussain committed:
ease arch constraints from x86 and ppc64le

Signed-off-by: Md. Shafi Hussain <Md.Shafi.Hussain@ibm.com>

Shafi-Hussain commented:
/build-konflux

grdryn commented:
> Could you please take a look for the arm64 image builds for jupyter-minimal-cpu-py311 and jupyter-minimal-cpu-py312?

@Shafi-Hussain could you please share direct links to the specific runs that have arm64 failures? It's difficult to determine because of all of the runs (including duplicates)

Shafi-Hussain commented:
> > Could you please take a look for the arm64 image builds for jupyter-minimal-cpu-py311 and jupyter-minimal-cpu-py312?
> 
> @Shafi-Hussain could you please share direct links to the specific runs that have arm64 failures? It's difficult to determine because of all of the runs (including duplicates)

Sure. 

[jupyter-minimal-cpu-py312 pipeline](https://konflux-ui.apps.stone-prd-rh01.pg1f.p1.openshiftapps.com/ns/rhoai-tenant/applications/automation/pipelineruns/odh-workbench-jupyter-minimal-cpu-py312-on-pull-request-pkvkf) failed on arm for some resource quota evaluation timeout.

[jupyter-minimal-cpu-py311 pipeline](https://konflux-ui.apps.stone-prd-rh01.pg1f.p1.openshiftapps.com/ns/rhoai-tenant/applications/automation/pipelineruns/odh-workbench-jupyter-minimal-cpu-py311-on-pull-request-mzjs6) is failing in ```ecosystem-cert-preflight-checks``` stage

Shafi-Hussain commented:
/build-konflux

grdryn commented:
> [jupyter-minimal-cpu-py312 pipeline](https://konflux-ui.apps.stone-prd-rh01.pg1f.p1.openshiftapps.com/ns/rhoai-tenant/applications/automation/pipelineruns/odh-workbench-jupyter-minimal-cpu-py312-on-pull-request-pkvkf) failed on arm for some resource quota evaluation timeout.

This one is a Konflux issue, not related to the build itself: Konflux just timed out provisioning the builder. No reason to think that it's reproducible. Indeed it can be seen on the new run that it has already passed that point.

> [jupyter-minimal-cpu-py311 pipeline](https://konflux-ui.apps.stone-prd-rh01.pg1f.p1.openshiftapps.com/ns/rhoai-tenant/applications/automation/pipelineruns/odh-workbench-jupyter-minimal-cpu-py311-on-pull-request-mzjs6) is failing in ecosystem-cert-preflight-checks stage

This is not specific to arm64: amd64 also has the same issue. I think if the checking of ppc64le didn't fail pulling the image, it would have failed on the same thing.

@jiridanek, are y'all aware of this issue in the preflight checks?
```json
"failed": [
            {
                "name": "HasLicense",
                "elapsed_time": 0,
                "description": "Checking if terms and conditions applicable to the software including open source licensing information are present. The license must be at /licenses",
                "help": "Check HasLicense encountered an error. Please review the preflight.log file for more information.",
                "suggestion": "Create a directory named /licenses and include all relevant licensing and/or terms and conditions as text file(s) in that directory.",
                "knowledgebase_url": "https://access.redhat.com/documentation/en-us/red_hat_software_certification/2024/html-single/red_hat_openshift_software_certification_policy_guide/index#assembly-requirements-for-container-images_openshift-sw-cert-policy-introduction",
                "check_url": "https://access.redhat.com/documentation/en-us/red_hat_software_certification/2024/html-single/red_hat_openshift_software_certification_policy_guide/index#assembly-requirements-for-container-images_openshift-sw-cert-policy-introduction"
            }
        ]
```

Shafi-Hussain committed:
Update pdf scripts

Updated install_texlive and install_pandoc scripts to run only on
ppc64le.
Other architectures use the default install_pdf_deps scripts.

This is to ensure s390x builds don't break with ppc64le scripts.

Signed-off-by: Md. Shafi Hussain <Md.Shafi.Hussain@ibm.com>

github-actions[bot] committed:
ci: update commit SHAs for image digests changes

Shafi-Hussain committed:
copy recursively

Signed-off-by: Md. Shafi Hussain <Md.Shafi.Hussain@ibm.com>

Shafi-Hussain committed:
Merge branch 'red-hat-data-services:rhoai-2.24' into rhoai-2.24

Shafi-Hussain committed:
change mount type from bind to cache

Signed-off-by: Md. Shafi Hussain <Md.Shafi.Hussain@ibm.com>

Shafi-Hussain commented:
/build-konflux

npanpaliya commented:
/build-konflux

jiridanek commented:
@coderabbitai wdyt?
